tests/pytests/functional/modules/test_mac_group.py

```diff
@@ -0,0 +1,193 @@
+"""
+    :codeauthor: Nicole Thomas <nicole@saltstack.com>
+"""
+
+import pytest
+from saltfactories.utils import random_string
+
+pytestmark = [
+    pytest.mark.slow_test,
+    pytest.mark.destructive_test,
+    pytest.mark.skip_if_not_root,
+    pytest.mark.skip_unless_on_darwin,
+]
+
+
+@pytest.fixture(scope="module")
+def group(modules):
+    return modules.group
+
+
+# Create group name strings for tests
+@pytest.fixture(scope="module")
+def add_group():
+    yield random_string("RS-", lowercase=False)
+
+
+@pytest.fixture(scope="module")
+def del_group():
+    yield random_string("RS-", lowercase=False)
+
+
+@pytest.fixture(scope="module")
+def change_group():
+    yield random_string("RS-", lowercase=False)
+
+
+@pytest.fixture(scope="module")
+def add_user():
+    yield random_string("RS-", lowercase=False)
+
+
+@pytest.fixture(scope="module")
+def rep_user_group():
+    yield random_string("RS-", lowercase=False)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def _setup_teardown_vars(group, add_group, change_group, del_group):
+    try:
+        yield
+    finally:
+        # Delete ADD_GROUP
+        add_info = group.info(add_group)
+        if add_info:
+            group.delete(add_group)
+
+        # Delete DEL_GROUP if something failed
+        del_info = group.info(del_group)
+        if del_info:
+            group.delete(del_group)
+
+        # Delete CHANGE_GROUP
+        change_info = group.info(change_group)
+        if change_info:
+            group.delete(change_group)
```

Comment:
```suggestion
@pytest.fixture(autouse=True)
def _setup_teardown_vars(group, add_group, change_group, del_group):
    try:
        yield
    finally:
        # Delete ADD_GROUP
        add_info = group.info(add_group)
        if add_info:
            group.delete(add_group)

        # Delete DEL_GROUP if something failed
        del_info = group.info(del_group)
        if del_info:
            group.delete(del_group)

        # Delete CHANGE_GROUP
        change_info = group.info(change_group)
        if change_info:
            group.delete(change_group)
```

This will make sure the system is "ready" and "clean" for each test.